### PR TITLE
Rewrite

### PR DESCRIPTION
**Pull request by andyjansson**
# Why?

The dependencies are poorly maintained and contains a lot of bugs. "Inhousing" the logic would allow us to better maintain the plugin and fix said bugs. 

# Internal changes

- Rewrote plugin to house all logic 🎉 
- Updated dependencies
    - Major changes: 
      - Upgraded to babel 7
      - Upgraded to ava 1.0.0-beta.8
- Updated targeted node versions in travis.yml

# Outward-facing changes

- Changed: Updated postcss-selector-parser to version 5.0.0-rc.3
- **Changed: Dropped reduce-css-calc as a dependency**
- Fixed: Support constant() and env() ([#42](https://github.com/postcss/postcss-calc/issues/42), [#48](https://github.com/postcss/postcss-calc/issues/48))
- Fixed: Support custom properties with "calc" in its name ([#50](https://github.com/postcss/postcss-calc/issues/50))
- Fixed: Remove unnecessary whitespace around `*` and `/` ([cssnano#625](https://github.com/cssnano/cssnano/issues/625))
- Fixed: Arithmetic bugs around subtraction ([#49](https://github.com/postcss/postcss-calc/issues/49))
- Fixed: Handling of nested calc statements ([reduce-css-calc#49](https://github.com/MoOx/reduce-css-calc/issues/49))
- Fixed: Bugs regarding complex calculations ([reduce-cs-calc#45](https://github.com/MoOx/reduce-css-calc/issues/45))
- Fixed: `100%` incorrectly being transformed to `1` ([reduce-css-calc#44](https://github.com/MoOx/reduce-css-calc/issues/44))
- Added: support for case-insensitive calc statements

### CODE REVIEW

**Comment by jonathantneal** [comment]
This looks beautifully written. I am 💯 for it. Not sure if anyone else needs to approve, but lemme ping @ai.

**Comment by ai** [comment]
@jonathantneal I gave @andyjansson access to this repo

**Comment by jonathantneal** [comment]
Great! @andyjansson, how can I help you?

**Comment by andyjansson** [comment]
>This looks beautifully written. I am 💯 for it. 

Glad you like it :)

>how can I help you?

Well, it would be a great help if anyone were willing to give it a spin before it goes live.

**Comment by evilebottnawi** [comment]
Just note, i i think will be great use `snapshots` for testing - easy write new tests, better reading, fast writing

**Comment by jonathantneal** [comment]
@andyjansson,

> https://github.com/andyjansson/postcss-calc/blob/master/src/lib/transform.js#L17

Is specifying the plugin necessary?

> ```
> (node:70308) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
> ```

Is this related to the tests + my version of node, or a third-party package issue, or something else?

**Comment by andyjansson** [comment]
>Is this related to the tests + my version of node, or a third-party package issue, or something else?

I saw this, too. I believe it's [an issue with yarn](https://github.com/yarnpkg/yarn/issues/5770), actually. It triggers even if you do something trivial, like `"hello": "echo hello"`. It is not present when executing it via npm.

File: package.json [hunk 1]
```diff
@@ -15,40 +15,45 @@
...
+    "eslint-plugin-import": "^2.2.0",
+    "jison-gho": "^0.6.1-215"
   },
   "dependencies": {
     "css-unit-converter": "^1.1.1",
     "postcss": "^7.0.2",
-    "postcss-selector-parser": "^2.2.2",
-    "reduce-css-calc": "^2.0.0"
+    "postcss-selector-parser": "^5.0.0-rc.3",
+    "postcss-value-parser": "^3.3.0"
```
[the remark this answers is not in the document]
Comment (andyjansson) [reply package.json]: ¯\\_(ツ)_/¯

I don't really use the parsing for anything else than picking out the `calc` function. The rest of the parsing is handled internally.

[the remark this answers is not in the document]
Comment (andyjansson) [reply package.json]: I'd prefer not to switch to `postss-values-parser` as it offers no real way of extracting the function for further parsing. It only supplies [a bunch of fragments](https://github.com/shellscape/postcss-values-parser/blob/master/test/function.js).

**Pull request merged**